### DISCUSSION TITLE
How to decode MUBUF type instructions

### DISCUSSION THREAD

**Discussion by Yzgll** [sarchlab/mgpusim#82] [Q&A]
I have added a new MUBUF instruction `24 BUFFER_STORE_BYTE: Untyped buffer store byte `to the decodetable, but there is no implementation of decodeMUBUF here https://github.com/sarchlab/mgpusim/blob/v3/insts/disassembler.go#L752, how do I implement it?
![p1](https://github.com/user-attachments/assets/eef729da-a471-4bae-9834-f54c6217cf23)

**Comment by syifan** [comment]
We have never implemented the support for MUBUF instructions. You will need to check the disassembler to see how to support it. This is not a very easy task. A better way to solve this problem is to slightly modify your kernel code to avoid this instruction. 

There is no plan to add extra support for GFX803 instruction set since it is no longer officially supported by ROCm. So I will close this issue.

**Reply by Yzgll** [reply]
Thanks for your reply, I have modified the kernel to avoid this instruction.
